lib/jsdom/living/structured-clone/structuredDeserializeWithTransfer.js

```diff
@@ -0,0 +1,448 @@
+     * for example, when both the source and target realms are in the same process.
+     */
+
+    if (transferDataHolder.Type === "ArrayBuffer") {
+      value = new ArrayBuffer(transferDataHolder.ArrayBufferByteLength);
+
+      const view = new Uint8Array(value);
+      view.set(transferDataHolder.ArrayBufferData, 0);
+    } else if (transferDataHolder.Type === "ResizableArrayBuffer") {
+    /**
+     * Otherwise, if transferDataHolder.[[Type]] is "ResizableArrayBuffer", then set value to a new ArrayBuffer
+     * object in targetRealm whose [[ArrayBufferData]] internal slot value is transferDataHolder.[[ArrayBufferData]],
+     * whose [[ArrayBufferByteLength]] internal slot value is transferDataHolder.[[ArrayBufferByteLength]],
+     * and whose [[ArrayBufferMaxByteLength]] internal slot value is transferDataHolder.[[ArrayBufferMaxByteLength]].
+     *
+     * For the same reason as the previous step, this step is also unlikely to throw an exception.
+     */
+
+      value = new ArrayBuffer(transferDataHolder.ArrayBufferByteLength, {
+        maxByteLength: transferDataHolder.ArrayBufferMaxByteLength
+      });
+
+      const view = new Uint8Array(value);
+      view.set(transferDataHolder.ArrayBufferData, 0);
+    } else {
+      /**
+       * Otherwise:
+       * Let interfaceName be transferDataHolder.[[Type]].
+       * If the interface identified by interfaceName is not exposed in targetRealm, then throw
+       * a "DataCloneError" DOMException.
+       * Set value to a new instance of the interface identified by interfaceName, created in targetRealm.
+       * Perform the appropriate transfer-receiving steps for the interface identified by interfaceName given
+       * transferDataHolder and value.
+       */
+
+      const interfaceName = transferDataHolder.Type;
+
+      /**
+       * If the interface identified by interfaceName is not exposed in targetRealm, then throw
+       * a "DataCloneError" DOMException.
+       */
+
+      if (!targetRealm[interfaceName]) {
+        throw DOMException.create(targetRealm, [
+          "The object can not be cloned.",
+          "DataCloneError"
+        ]);
+      }
+
+      /**
+       * Set value to a new instance of the interface identified by interfaceName, created in targetRealm.
+       */
+
+      value = Reflect.construct(targetRealm[interfaceName], []);
+    }
+
+    /**
+     * Set memory[transferDataHolder] to value.
+     * Append value to transferredValues.
+     */
+
+    memory[transferDataHolder] = value;
+    transferredValues.push(value);
+  }
+
+  /**
+   * Let deserialized be ? StructuredDeserialize(serializeWithTransferResult.[[Serialized]], targetRealm, memory).
+   * Return { [[Deserialized]]: deserialized, [[TransferredValues]]: transferredValues }.
+   */
+
+  const deserialized = structuredDeserialize(serializeWithTransferResult.Serialized, targetRealm, memory);
+
+  return {
+    Deserialized: deserialized,
```

Comment:
Here and elsewhere, let's prefer sticking to JavaScript conventional property names instead of spec ones. So, lowerCamelCase.